libraries/AP_HAL_ChibiOS/RCOutput.cpp

```diff
@@ -253,14 +253,22 @@ void RCOutput::dshot_collect_dma_locks(uint32_t time_out_us)
             // if we have time left wait for the event
             eventmask_t mask = 0;
             const uint32_t pulse_elapsed_us = now - group.last_dmar_send_us;
+            uint32_t wait_us = 0;
             if (now < time_out_us) {
-                mask = chEvtWaitOneTimeout(group.dshot_event_mask,
-                    chTimeUS2I(MAX(time_out_us - now, group.dshot_pulse_send_time_us - pulse_elapsed_us)));
+                wait_us = MAX(time_out_us - now, group.dshot_pulse_send_time_us - pulse_elapsed_us);
             } else if (pulse_elapsed_us < group.dshot_pulse_send_time_us) {
                 // better to let the burst write in progress complete rather than cancelling mid way through
-                mask = chEvtWaitOneTimeout(group.dshot_event_mask,
-                    chTimeUS2I(group.dshot_pulse_send_time_us - pulse_elapsed_us));
+                wait_us = group.dshot_pulse_send_time_us - pulse_elapsed_us;
             }
+
+            // waiting for a very short period of time can cause a
+            // timer wrap with ChibiOS timers. Use CH_CFG_ST_TIMEDELTA
+            // as minimum. Don't allow for a very long delay (over 1200us)
+            // to prevent bugs in handling timer wrap
+            const uint32_t max_delay_us = 1200;
+            wait_us = constrain_uint32(wait_us, CH_CFG_ST_TIMEDELTA, max_delay_us);
```

Comment:
so I think the correct value here is _dshot_period_us (rather than 1200) - we should try that first

Reply:
i don't think so - we're interested in the max time we're willing to wait, which could be longer than dshot period with DMA contention